snooty/postprocess.py

```diff
@@ -2193,23 +2195,162 @@ def exit_page(self, fileid_stack: FileIdStack, page: Page) -> None:
 
 
 class ComposableTutorialHandler(Handler):
-    """Handles composable tutorial directivepresence in page.
-    Should not be simultaneously present on page with other directives"""
+    """Handles composable tutorial directive presence in page.
+    Should not be simultaneously present on page with other directives
+    Checks presence of selected-content directive in composable-tutorial"""
+
+    def handle_composable_content(
+        self, fileid_stack: FileIdStack, node: n.ComposableContent
+    ) -> None:
+        # if no composable_tutorial found before composable content throw an error
+        if not self.composable_tutorial:
+            self.context.diagnostics[fileid_stack.current].append(
+                UnexpectedDirectiveOrder(
+                    "composable-tutorial must be present on the page before selected-content",
+                    node.span[0],
+                )
+            )
+            return
+
+        # validate selection ids by checking against composable tutorial and spec
+        try:
+            selection_ids = util.split_option_str(node.options.get("selections", ""))
+        except ValueError:
+            self.context.diagnostics[fileid_stack.current].append(
+                InvalidChildCount(
+                    "selected-content",
+                    "selections",
+                    "at least one",
+                    node.span[0],
+                )
+            )
+            return
+
+        selections: Dict[str, str] = {}
+        for idx in range(len(selection_ids)):
+            selection_id = selection_ids[idx]
+            try:
+                composable_option = self.composable_tutorial.composable_options[idx]
```

Comment:
I bet this is right, but just checking that we shouldn't be using `selection_id` rather than `idx` here on line 2233?

Reply:
left a comment!